src/core/api/commissioner_api.cpp

```diff
@@ -40,14 +40,14 @@
 
 using namespace ot;
 
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
```

Comment:
I think it would be simpler to have `OPENTHREAD_CONFIG_COMMISSIONER_ENABLE` here and check for `OPENTHREAD_FTD` in `openthread-core-config-check.h`. Thoughts?

Reply:
I think when COMMISSIONER=1 and --enable-ftd and --enable-mtd, both openthread_ftd.a and openthread_mtd.a will be compiled with OPENTHREAD_CONFIG_COMMISSIONER_ENABLE=1.